src/Handlers/UserHandler.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Psalm\LaravelPlugin\Handlers;
+
+use Psalm\Plugin\EventHandler\Event\MethodReturnTypeProviderEvent;
+use Psalm\Plugin\EventHandler\MethodReturnTypeProviderInterface;
+use Psalm\Type;
+
+final class UserHandler implements MethodReturnTypeProviderInterface
+{
+    public static function getClassLikeNames(): array
+    {
+        return [
+            \Illuminate\Support\Facades\Auth::class,
+            'Auth',
+            \Illuminate\Http\Request::class
+        ];
+    }
+
+    public static function getMethodReturnType(
+        MethodReturnTypeProviderEvent $event
+    ): ?Type\Union {
+        $method_name_lowercase = $event->getMethodNameLowercase();
+        $class_name = $event->getFqClasslikeName();
+
+        if (
+            $class_name === \Illuminate\Support\Facades\Auth::class ||
+            $class_name === 'Auth'
+        ) {
+            if ($method_name_lowercase === 'user') {
+                return new Type\Union([
+                    new Type\Atomic\TNamedObject('App\Models\User'),
```

Comment:
This does work for the _default_ laravel install, i wonder if we could dynamically look up what object to use 🤔 

I have lots of projects that have a different namespace etc. thinking out loud

Reply:
Well two things I suppose (and overall I agree with you).

- In #184 you can declare the model but that wouldn't work in my case (where in my code I've added my own authprovider)
- If there is a place to set config (I suppose Psalm.xml) for this that would be the best place to set it I think as this plugin does not read anything for itself from Laravel's config subsystem

Reply:
> If there is a place to set config (I suppose Psalm.xml) for this that would be the best place to set it I think as this plugin does not read anything for itself from Laravel's config subsystem

I wouldn't be _opposed_ to doing this -- we rely on dynamic analysis in order to fetch container dependencies etc. Whether it's possible or not to determine the correct provider to use, i dunno off the top of my head